### PR TITLE
Adds public interface to apply filters

### LINKED ISSUE #48: Add public, layer-level interface for filtering
Need to discuss this with @alonsogarciapablo

### PR DESCRIPTION
**Pull request by fdansv**
Closes #48 

@alonsogarciapablo This is compatible with all the requisites you asked for the integration with cartodb.js + deep-insights

### CODE REVIEW

File: src/filter.js [hunk 1]
```diff
@@ -96,6 +99,12 @@ cartodb.d3.extend(Filter.prototype, cartodb.d3.Event, {
 })
 
 Filter.accept = function (terms) {
+  if (terms === 'all') {
+    return function () { return true }
+  }
+  else if (terms = 'none'){
```
Comment (pabletecodes) [inline src/filter.js]: `terms === 'none'`?

File: src/filter.js [hunk 2]
```diff
@@ -104,10 +113,17 @@ Filter.accept = function (terms) {
...
     }
+    return false
   }
 }
 
 Filter.reject = function (terms) {
+  if (terms === 'all') {
+    return function () { return false }
+  }
+  else if (terms = 'none'){
```
Comment (pabletecodes) [inline src/filter.js]: `terms === 'none'`?

File: src/filter.js (hunk 1 above)
Comment (pabletecodes) [inline src/filter.js]: Also this line should be merged with the previous line

Comment (fdansv) [inline src/filter.js]: omg

**Pull request merged**